### PR TITLE
Add tree type column to phylo trees view

### PR DESCRIPTION
**Pull request by jgadling**
### Description

Add a tree type column to the phylo trees data table

#### Issue
[ch158247](https://app.clubhouse.io/genepi/story/158247)

### Test plan

Tested in local dev and it looks good, I'll add a link to the rdev site once I have it.

### CODE REVIEW

File: src/frontend/src/common/components/library/data_table/style.ts [hunk 1]
```diff
@@ -73,3 +84,20 @@ export const HeaderCheckbox = styled(Checkbox)`
...
+export interface AlignProps extends Props {
+  align?: string;
+}
+
+export const TableHeader = styled("div", {
+  shouldForwardProp: (prop) => !doNotForwardProps.includes(prop as string),
+})`
+  ${(props: AlignProps) => {
+    const { align } = props;
+    const justify = align !== undefined ? align : "left";
```
Comment (mmmmmmaya) [inline src/frontend/src/common/components/library/data_table/style.ts]:
```suggestion
    const justify = align ?? "left";
```

Comment (mmmmmmaya) [reply src/frontend/src/common/components/library/data_table/style.ts]: This is the TS way of saying "set the value to variable, but if it doesn't exist use this value instead"

File: src/frontend/src/common/components/library/data_table/style.ts [hunk 2]
```diff
@@ -16,15 +16,26 @@ export const TableRow = styled.div`
...
+  shouldForwardProp: (prop) => !doNotForwardProps.includes(prop as string),
+})`
   display: flex;
   align-items: center;
   width: 100%;
 
-  ${(props) => {
+  ${(props: ExtraProps) => {
+    const { header } = props;
+    const align = header && header.align !== undefined ? header.align : "left";
```
Comment (mmmmmmaya) [inline src/frontend/src/common/components/library/data_table/style.ts]:
```suggestion
    const align = header?.align ?? "left";
```

File: src/frontend/src/common/components/library/data_table/style.ts [hunk 3]
```diff
@@ -73,3 +84,20 @@ export const HeaderCheckbox = styled(Checkbox)`
     `;
   }}
 `;
+
+export interface AlignProps extends Props {
+  align?: string;
+}
+
+export const TableHeader = styled("div", {
+  shouldForwardProp: (prop) => !doNotForwardProps.includes(prop as string),
```
Comment (phoenixAja) [inline src/frontend/src/common/components/library/data_table/style.ts]: do you still need this line? it looks like TableHeader doesn't have a `header` prop?

**Comment by phoenixAja** [comment]
@jgadling awesome work this is really exciting!!! do you have the rdev link yet? Also @mmmmmmaya does this PR fix the table alignment issue on narrow screens we were looking at earlier?

**Comment by jgadling** [comment]
> @jgadling awesome work this is really exciting!!! do you have the rdev link yet? Also @mmmmmmaya does this PR fix the table alignment issue on narrow screens we were looking at earlier?

Rdev link: https://treetype-frontend.dev.genepi.czi.technology

**Review by mmmmmmaya — APPROVED** [review]

**Comment by mmmmmmaya** [comment]
@phoenixAja Fortunately the trees table does not suffer from this as much as the sample table (in fact, I could not see the problem at all on the trees table, even on small screens). So I'm not sure if it is fixed or not, but it will give us some ideas for the future when we do attempt to address this problem on the samples table

**Pull request merged**